plonk-core/src/circuit.rs

```diff
@@ -280,7 +280,7 @@ where
         let mut prover = Prover::new(transcript_init);
         // Fill witnesses for Prover
         self.gadget(prover.mut_cs())?;
-        prover.cs.public_inputs.update_size(circuit_size);
+        prover.cs.public_inputs.update_size(prover.circuit_bound());
```

Comment:
Why not adding a fn for Prover like: 
`Prover::update_pi_size` for which the prover has the `circuit_bound` already. And we don't need to pass any parameters.

Reply:
While trying this out I realized we may as well try to get rid of `n` in the `PI`. I'll open a separate PR for this.